src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Symbols/SymbolManagerBase.cs

```diff
@@ -328,26 +328,34 @@ public TypeArrayKey(CType[] types)
 
             public bool Equals(TypeArrayKey other)
             {
-                if (other._types == _types) return true;
-                if (other._types.Length != _types.Length) return false;
-                if (other._hashCode != _hashCode) return false;
-                for (int i = 0; i < _types.Length; i++)
+                CType[] types = _types;
+                CType[] otherTypes = other._types;
+                if (other._types == _types)
                 {
-                    if (!_types[i].Equals(other._types[i]))
+                    return true;
+                }
+
+                if (other._hashCode != _hashCode || otherTypes.Length != types.Length)
+                {
+                    return false;
+                }
+
+                for (int i = 0; i < otherTypes.Length; i++)
```

Comment:
I don't think it matter, but I'm curious why you changed this from types to otherTypes.